src/routes/route.py

```diff
@@ -1,67 +1,95 @@
+import logging
+
 import geopandas as gpd
+import numpy as np
+from numpy.typing import NDArray
 from tqdm import tqdm
 
 from data_loader.population.utils import NODE_ID, POPULATION
 from routes.route_utils import path
 
 
 class Route:
-    def __init__(self, route_path: path, num_people_on_route: int) -> None:
+    def __init__(
+        self, route_path: path, num_people_on_route: int, departure_times: list[int]
+    ) -> None:
+        if len(departure_times) != num_people_on_route:
+            logging.fatal(
+                "Number of departure times must equal the number of people on route."
+            )
```

Comment:
This fatal error message is not actually fatal.

```suggestion
        if len(departure_times) != num_people_on_route:
            logging.fatal(
                "Number of departure times must equal the number of people on route."
            )
            raise ValueError("Mismatch between departure times and number of people on route.")
```